### PR TITLE
[bridge]  support rank validators by voting power when requesting signatures

### PR DESCRIPTION
**Pull request by longbowlu**
## Description 

In this PR we introduce a way to order validators to request signatures according to their voting power. This is useful for signatures to be verified on ethereum because the # of signatures do impact the gas cost there significantly.

## Test plan 

deploying on a testnet bridge node to test.

---

## Release notes

Check each box that your changes affect. If none of the boxes relate to your changes, release notes aren't required.

For each box you select, include information after the relevant heading that describes the impact of your changes that a user might notice and any actions they must take to implement updates. 

- [ ] Protocol: 
- [ ] Nodes (Validators and Full nodes): 
- [ ] Indexer: 
- [ ] JSON-RPC: 
- [ ] GraphQL: 
- [ ] CLI: 
- [ ] Rust SDK:

### CODE REVIEW

**Comment by patrickkuo** [comment]
Will this change cause more traffic to the same validators, will that cause congestion?

File: crates/sui-bridge/src/client/bridge_authority_aggregator.rs [hunk 1]
```diff
@@ -163,9 +164,25 @@ async fn request_sign_bridge_action_into_certification(
...
+    // to save gas cost.
+    let preference = if matches!(action, BridgeAction::SuiToEthBridgeAction(_)) {
+        Some(BTreeSet::new())
+    } else if matches!(action, BridgeAction::EthToSuiBridgeAction(_))
+        || action.chain_id().is_sui_chain()
+    {
+        None
+    } else {
+        Some(BTreeSet::new())
+    };
```
Comment (dariorussi) [inline crates/sui-bridge/src/client/bridge_authority_aggregator.rs:181]: I find this condition very difficult to read, and I guess is the `matches` used with `if` or the `||` condition. Not sure.
In any case is this the same as 
```
let preferences = match action {
  BridgeAction::SuiToEthBridgeAction(_) => Some(BTreeSet::new()),
  BridgeAction::EthToSuiBridgeAction(_) => None,
  _ =>  {
    if action.chain_id().is_sui_chain() {
      None
    } else {
      Some(BTreeSet::new())
  }
}

```

File: crates/sui-bridge/src/types.rs [hunk 2]
```diff
@@ -116,18 +116,19 @@ impl BridgeCommittee {
...
+    // Note: blocklisted members are always excluded.
     fn shuffle_by_stake_with_rng(
         &self,
-        // preference is not supported today
-        _preferences: Option<&BTreeSet<BridgeAuthorityPublicKeyBytes>>,
+        // `preferences` is used as a flag here to influence the order of validators to be requested
+        //  if `Some(_)`, then we will request validators in the order of the voting power
+        //  if `None`, we still refer to voting power, but they are shuffled by randomness.
+        //  to save gas cost.
+        preferences: Option<&BTreeSet<BridgeAuthorityPublicKeyBytes>>,
```
Comment (dariorussi) [inline crates/sui-bridge/src/types.rs:126]: is this truly used as a flag? why the `BTreeSet`? are we going to use it later? I am a bit lost here :)

**Review by dariorussi — APPROVED** [review]
seems good to me

**Comment by dariorussi** [comment]
please take a look at @patrickkuo's question

File: crates/sui-bridge/src/types.rs (hunk 2 above)
Comment (longbowlu) [reply crates/sui-bridge/src/types.rs:126]: this is a trait interface. Here we are indeed using it as a boolean to indicate if we want to order it in one way or another (starting from line 150)

**Comment by longbowlu** [comment]
> Will this change cause more traffic to the same validators, will that cause congestion?

the impact should be minimal since the congestion is mostly on the eth end. While we apply this to `SuiToEthBridgeAction` and eth governance actions. The former queries sui as opposed to eth, and the latter's rare

**Pull request merged**